lib/streamlit/echo.py

```diff
@@ -58,22 +58,24 @@ def echo(code_location="above"):
         # Get stack frame *before* running the echoed code. The frame's
         # line number will point to the `st.echo` statement we're running.
         frame = traceback.extract_stack()[-3]
-        filename, start_line = frame.filename, frame.lineno
+        filename, start_line = frame.filename, frame.lineno or 0
```

Comment:
we need this `or 0` part to make mypy happy, since `lineno` is optional, but in our case, it always exists